qiskit/circuit/instruction.py

```diff
@@ -81,18 +82,27 @@ def __eq__(self, other):
         Returns:
             bool: are self and other equal.
         """
-        res = False
-        if type(self) is type(other) and \
-                self.name == other.name and \
-                self.num_qubits == other.num_qubits and \
-                self.num_clbits == other.num_clbits and \
-                self.definition == other.definition and \
-                (self.params == other.params or
-                 numpy.allclose([float(p) for p in self.params],
-                                [float(p) for p in other.params],
-                                atol=_CUTOFF_PRECISION)):
-            res = True
-        return res
+        if type(self) is not type(other) or \
+                self.name != other.name or \
+                self.num_qubits != other.num_qubits or \
+                self.num_clbits != other.num_clbits or \
+                self.definition != other.definition:
+            return False
+
+        for self_param, other_param in zip_longest(self.params, other.params):
```

Comment:
couldn't we just return False in the previous block if `len(self.params) != len(other.params)`, and here use a regular zip?